### DISCUSSION TITLE
Ctrl+Backspace should delete the rest of the word left of the cursor in Insert mode

### DISCUSSION THREAD

**Discussion by jroth5** [helix-editor/helix#6971] [Ideas]
Ctrl+Backspace is a pretty common shortcut and it would be nice for it to work in insert mode. It deletes everything from the left of the cursor to the start of the current word, placing the cursor where the last character used to be. Currently, ctrl+backspace does nothing.

**Comment by CptPotato** [comment]
<kbd>Alt</kbd> <kbd>Backspace</kbd> works like this.

I think this is common on Linux (<kbd>Ctrl</kbd> <kbd>Backspace</kbd> is usually found on Windows).

**Reply by jroth5** [reply]
Ahh, never knew that.

I tried adding this to my config:
`[keys.insert]`
`C-backspace = "delete-word-backward"`

but it gets ignored (doing a normal backspace instead). Anyone know why that is? Is there some other way to remap it?

**Reply by kirawi** [reply]
Might be eaten by your terminal. You can try Wezterm which supports the extended keyboard protocol.

**Reply by the-mikedavis** [reply]
Yeah I believe that's the case: `C-backspace` sends the same escape codes as `C-h`. In order to bind this you need a terminal that supports and enables the enhanced keyboard protocol: https://github.com/helix-editor/helix/wiki/Terminal-Support#enhanced-keyboard-protocol

**Comment by kobakt** [comment]
I just found out that both of these commands already work in command mode and in the file picker (on windows), but not in insert mode. It was a little confusing when they didn't work when I thought I had already been using them. 

While it's not hard to add these as keybinds, for consistency's sake maybe they should just be added as extra defaults.

**Comment by KySpace** [comment]
Is there a command to delete till next word's beginning? Now I bind the C-del to `delete_word_forward` but that doesn't quite behave like usual typing. For example I have a lot of white space after the cursor until the next word, and I want to delete those, currently this will delete the next word as well, while I just want to remove the next space.
